ecs-agent/netlib/platform/firecracker_linux.go

```diff
@@ -65,7 +68,21 @@ func (f *firecraker) ConfigureInterface(
 	iface *networkinterface.NetworkInterface,
 	netDAO netlibdata.NetworkDataClient,
 ) error {
-	return f.common.configureInterface(ctx, netNSPath, iface, netDAO)
+	var err error
+	switch iface.InterfaceAssociationProtocol {
+	case networkinterface.DefaultInterfaceAssociationProtocol:
+		err = f.common.configureRegularENI(ctx, netNSPath, iface)
+	case networkinterface.VLANInterfaceAssociationProtocol:
+		err = f.configureBranchENI(ctx, netNSPath, iface)
+	case networkinterface.V2NInterfaceAssociationProtocol:
+		err = f.common.configureGENEVEInterface(ctx, netNSPath, iface, netDAO)
+	case networkinterface.VETHInterfaceAssociationProtocol:
+		// Do nothing.
```

Comment:
Can you add a comment for why doing nothing is the right thing to be doing here?

Reply:
Ack. Will do.

Reply:
created a new PR with the findings, I screwed up the commit history when I attempted to rebase/squash
https://github.com/aws/amazon-ecs-agent/pull/4436